yt_dlp/downloader/hls.py

```diff
@@ -83,14 +83,13 @@ def real_download(self, filename, info_dict):
 
         if not self.can_download(s, info_dict, self.params.get('allow_unplayable_formats')):
             if info_dict.get('extra_param_to_segment_url') or info_dict.get('_decryption_key_url'):
-                self.report_error('pycryptodome not found. Please install it.')
+                self.report_error('[%s] pycryptodome not found. Please install it.' % self.FD_NAME)
                 return False
             if self.can_download(s, info_dict, with_crypto=True):
-                self.report_warning('pycryptodome is needed to download this file with hlsnative')
-            self.report_warning(
-                'hlsnative has detected features it does not support, '
-                'extraction will be delegated to ffmpeg')
+                self.report_warning('[%s] pycryptodome is needed to download this file' % self.FD_NAME)
             fd = FFmpegFD(self.ydl, self.params)
+            self.report_warning(
+                '[%s] Detected unsupported features, extraction will be delegated to %s' % fd.get_basename())
```

Comment:
Two `%s` but only one variable

Reply:
Yes, thank you for pointing that out. I fix it